cli-tools/sagebig/src/main.rs

```diff
@@ -97,3 +113,32 @@ fn cmd_contains(args: &clap::ArgMatches) -> Result<(), ReadError> {
 
     Ok(())
 }
+
+fn cmd_extract(args: &clap::ArgMatches) -> Result<(), ReadError> {
+    let big_path = args.value_of("archive_path").unwrap();
+    let archive = try!(BigArchive::new_from_path(&big_path));
+    let output_dir = Path::new(args.value_of("output_dir").unwrap());
+
+    // Create output dir if it doesn't exist
+    if !output_dir.exists() {
+        try!(fs::create_dir(&output_dir));
+        println!("Created output_dir `{:?}`", &output_dir.display());
+    }
+
+    let output_dir = output_dir.canonicalize().unwrap();
+
+    for name in archive.get_all_entry_names() {
+        let entry = archive.get_entry(name)
+            .expect(&format!("Failed to read known entry {} from {}", name, big_path));
+
+        let file_path = output_dir.join(Path::new(&entry.name));
+        let file_dirpath = &file_path.parent().unwrap();
+
+        if !file_dirpath.exists() {
+            fs::DirBuilder::new().recursive(true).create(&file_dirpath).unwrap();
+            println!("Created path `{}`", &file_dirpath.display());
```

Comment:
Would you kindly put these `println!`s behind a `-v|--verbose` flag?

hint: https://docs.rs/clap/2.20.0/clap/struct.ArgMatches.html#method.is_present